craft_providers/bases/almalinux.py

```diff
@@ -211,6 +211,9 @@ def _setup_snapd(self, executor: Executor) -> None:
                 details=details_from_called_process_error(error),
             ) from error
 
+        self._enable_snapd_service(executor=executor)
```

Comment:
So if I understand this correctly, changing these calls from `_post_setup_snapd()` to `_setup_snapd` had the consequence of every subclass of `Base` needing to repeat these calls in their implementation of `_setup_snapd`, right? 

I agree with the overall intent of doing the hold only once, but this repetition is unfortunate because it's the kind of thing that's easy to miss when a new base comes along, or if we need to refactor this (will need to remember to update all current implementations). I don't have a "great" idea to address this, the only one that comes to mind is a new method (like `_configure_snapd()`) that gets called between the `_setup_snapd` and `_post_setup_snapd` and doesn't get called in warmup.

Also it was a bit unclear to understand that `_post_setup_snapd()` is called both at setup() and at warmup(). Can we update its docstring to make this more evident?

Reply:
I think repeat call `snap refresh --hold` shouldn't have issue? So keep them in the `_post_setup_snapd` should be OK?

Reply:
Yeah, I think I shouldn't have tried to do 2 things in 1 PR (use the new call and speed up the warmup time).

I agree that this approach is more error prone and I don't have a good idea either.  Perhaps `warmup` should make a few direct calls (`setup_snapd_proxy()`) instead of calling the high-level setup steps (`post_setup_snapd`).

Let's talk during standup tomorrow.

Reply:
Thanks for your patience with explaining, I have added what we discussed today.  It's made the PR cleaner and improved warmup time.  

I'm still thinking about a future improvement for `warmup`.  I'm wondering about a series of warmup calls like:
```
def warmup():
  _warmup_image()
  _warmup_snapd()
  _warmup_snaps()
```